galsim/wfirst/wfirst_psfs.py

```diff
@@ -172,269 +229,114 @@ def getPSF(SCAs=None, approximate_struts=False, n_waves=None, extra_aberrations=
             oversampling = 1.5
         else:
             oversampling = 1.2
-            pupil_plane_im = galsim.wfirst.pupil_plane_file
-            pupil_plane_scale = galsim.wfirst.pupil_plane_scale
-
-    if wavelength is None:
-        if n_waves is not None:
-            if wavelength_limits is None:
-                # To decide the range of wavelengths to use (if none were passed in by the user),
-                # first check out all the bandpasses.
-                bandpass_dict = galsim.wfirst.getBandpasses()
-                # Then find the blue and red limit to be used for the imaging bandpasses overall.
-                blue_limit, red_limit = _find_limits(default_bandpass_list, bandpass_dict)
-            else:
-                if not isinstance(wavelength_limits, tuple):
-                    raise TypeError("Wavelength limits must be entered as a tuple.")
-                blue_limit, red_limit = wavelength_limits
-                if red_limit <= blue_limit:
-                    raise galsim.GalSimIncompatibleValuesError(
-                        "Wavelength limits must have red_limit > blue_limit.",
-                        blue_limit=blue_limit, red_limit=red_limit)
-    elif isinstance(wavelength, float):
-        wavelength_nm = wavelength
-    elif isinstance(wavelength, galsim.Bandpass):
-        wavelength_nm = wavelength.effective_wavelength
-    else:
-        raise TypeError("wavelength should either be a Bandpass, float, or None.")
-
-    # Start reading in the aberrations for the relevant SCAs.
-    aberration_dict = {}
-    PSF_dict = {}
-    if logger: logger.debug('Beginning to loop over SCAs and get the PSF:')
-    for SCA in SCAs:
-        aberration_dict[SCA] = _read_aberrations(SCA)
-
-        use_aberrations = aberration_dict[SCA]
-        if extra_aberrations is not None:
-            use_aberrations += extra_aberrations
-        # We don't want to use piston, tip, or tilt aberrations.  The former doesn't affect the
-        # appearance of the PSF, and the latter cause centroid shifts.  So, we set the first 4
-        # numbers (corresponding to a place-holder, piston, tip, and tilt) to zero.
-        use_aberrations[0:4] = 0.
-
-        # Now set up the PSF for this SCA, including the option to simplify the pupil plane.
-        if logger: logger.debug('   ... SCA %d'%SCA)
-        if wavelength is None:
-            if approximate_struts:
-                PSF = galsim.ChromaticOpticalPSF(
-                    lam=zemax_wavelength,
-                    diam=galsim.wfirst.diameter, aberrations=use_aberrations,
-                    obscuration=galsim.wfirst.obscuration, nstruts=6,
-                    oversampling=oversampling, gsparams=gsparams)
+            if pupil_plane_type == 'long':
+                pupil_plane_im = galsim.wfirst.pupil_plane_file_longwave
```

Comment:
Also not covered.  This one is the fastest case, so we should probably actually check it somehow.  Maybe as simple as just checking that the pupil planes generated for Z and F are different.